ci/run_tests.sh

```diff
@@ -13,86 +13,70 @@ fi
 pushd example_packages/
 rm -rf ./*/build
 
-pushd hello_world
-"$fpm" build
-"$fpm" run --target hello_world
-"$fpm" run
-popd
-
-pushd hello_fpm
-"$fpm" build
-"$fpm" run --target hello_fpm
-popd
-
-pushd circular_test
-"$fpm" build
-popd
-
-pushd circular_example
-"$fpm" build
-popd
-
-pushd hello_complex
-"$fpm" build
-"$fpm" test
-"$fpm" run --target say_Hello
-"$fpm" run --target say_goodbye
-"$fpm" test --target greet_test
-"$fpm" test --target farewell_test
-popd
-
-pushd hello_complex_2
-"$fpm" build
-"$fpm" run --target say_hello_world
-"$fpm" run --target say_goodbye
-"$fpm" test --target greet_test
-"$fpm" test --target farewell_test
-popd
-
-pushd with_examples
-"$fpm" build
-"$fpm" run --example --target demo-prog
-"$fpm" run --target demo-prog
-popd
-
-pushd auto_discovery_off
-"$fpm" build
-"$fpm" run --target auto_discovery_off
-"$fpm" test --target my_test
-test ! -x ./build/gfortran_*/app/unused
-test ! -x ./build/gfortran_*/test/unused_test
-popd
-
-pushd with_c
-"$fpm" build
-"$fpm" run --target with_c
-popd
-
-pushd submodules
-"$fpm" build
-popd
-
-pushd program_with_module
-"$fpm" build
-"$fpm" run --target Program_with_module
-popd
-
-pushd link_executable
-"$fpm" build
-"$fpm" run --target gomp_test
-popd
-
-pushd fortran_includes
-"$fpm" build
-popd
-
-pushd c_includes
-"$fpm" build
-popd
-
-pushd c_header_only
-"$fpm" build
-popd
+dir=hello_world
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target hello_world
+"$fpm" -C $dir run
+
+dir=hello_fpm
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target hello_fpm
+
+dir=circular_test
+"$fpm" -C $dir build
+
+dir=circular_example
+"$fpm" -C $dir build
+
+dir=hello_complex
+"$fpm" -C $dir build
+"$fpm" -C $dir test
+"$fpm" -C $dir run --target say_Hello
+"$fpm" -C $dir run --target say_goodbye
+"$fpm" -C $dir test --target greet_test
+"$fpm" -C $dir test --target farewell_test
+
+dir=hello_complex_2
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target say_hello_world
+"$fpm" -C $dir run --target say_goodbye
+"$fpm" -C $dir test --target greet_test
+"$fpm" -C $dir test --target farewell_test
+
+dir=with_examples
+"$fpm" -C $dir build
+"$fpm" -C $dir run --example --target demo-prog
+"$fpm" -C $dir run --target demo-prog
+
+dir=auto_discovery_off
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target auto_discovery_off
+"$fpm" -C $dir test --target my_test
+test ! -x $dir/build/gfortran_*/app/unused
+test ! -x $dir/build/gfortran_*/test/unused_test
+
+dir=with_c
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target with_c
+
+"$fpm" -C $dir build
+
+dir=program_with_module
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target Program_with_module
+
+dir=link_executable
+"$fpm" -C $dir build
+"$fpm" -C $dir run --target gomp_test
+
+dir=fortran_includes
+"$fpm" -C $dir build
+
+dir=c_includes
+"$fpm" -C $dir build
+
+dir=c_header_only
+"$fpm" -C $dir build
```

Comment:
I am not sure about these changes. I feel the previous code was simpler? Also the previous code tests the "default" usage of fpm, so I think we should keep it.

The `-C` should also be tested, but I think most people will use `fpm` without `-C`, so that should be tested by default I think.

Reply:
I don't mind this change, it's a little more concise and aligns with how other tools might be batch called with `-C`. It is true that we still need to test the default usage as well though.

Reply:
Changed, only one package will test the directory change and automatic discovery now.